packages/salesforcedx-apex-replay-debugger/src/states/variableAssignmentState.ts

```diff
@@ -44,7 +44,11 @@ export class VariableAssignmentState implements DebugLogState {
       if (logContext.getStaticVariablesClassMap().has(className)) {
         map = logContext.getStaticVariablesClassMap().get(className)!;
         container = map.get(varName)! as ApexVariableContainer;
-      } else if (frameInfo.locals.has(varName)) {
+        // If the className is 'this' that means the variable being split was
+        // this.<something>. We need to check the className for 'this' otherwise
+        // a propery on 'this' would get incorrectly processed as a local variable.
+        // It'll correclty fall into the local variable
+      } else if (className !== 'this' && frameInfo.locals.has(varName)) {
```

Comment:
To be clear this was tested to levels of indirection as well. this.myInteger with myInteger as a parameter as well as this.myInnerClass.myInteger with myInteger is a parameter.